services/wallet/api.go

```diff
@@ -312,14 +315,14 @@ func (api *API) GetEthereumChains(ctx context.Context, onlyEnabled bool) ([]*par
 	return api.s.rpcClient.NetworkManager.Get(onlyEnabled)
 }
 
-func (api *API) FetchPrices(ctx context.Context, symbols []string, currency string) (map[string]float64, error) {
+func (api *API) FetchPrices(ctx context.Context, symbols []string, currencies []string) (map[string]map[string]float64, error) {
```

Comment:
I am trying to understand why this parameter changed from a single currency to multiple currencies. In which case do we need results in various currencies?

Reply:
- We always need the price in USD to calculate how many decimals we display for a certain token, regardless of the currency chosen by the user in the settings. For this reason, the underlying function call (`fetchCryptoComparePrices`) now takes multiple currencies, so all the prices can be retrieved in a single CryptoCompare request.
- I had the option of keeping `FetchPrices` with a single currency (and pass a vector with that single currency to `fetchCryptoComparePrices`) or extending the flexibility of multiple currencies to it as well. I went with the second option since it could eventually help condense several API calls into one.
- John mentioned in a call that in the future the user might want to select multiple user currencies instead of a single one, and he should be able to quickly switch between them (for example, in high-inflation countries people normally "think" in their local currency and some more stable one -like USD- at the same time). In that case, we'll want to always fetch values for all chosen currencies and not have to re-fetch (with its corresponding delay) when the user decides to switch between them. I did this change to "future-proof" the Go API a little bit.

Does this clarify things a little bit? Does this decision make sense now?

Reply:
Thanks, @dlipicar, for the excellent clarifications, and I support your decision.